main/mock/framework.contracts.ts

```diff
@@ -0,0 +1,28 @@
+// We copy types from frameworks here so we have no dependency on jest, jasmine or vitest when running in a consuming project that will not have types for these frameworks.
+
```

Comment:
we should probably have version numbers etc so at least we know our compatibility

Reply:
the version numbers are defined in package.json as optional peers. If these types are not compatible they will throw and error at compile time.